core/src/main/java/io/grpc/Context.java

```diff
@@ -49,13 +46,11 @@
 import javax.annotation.Nullable;
 
 /**
- * A context propagation mechanism which carries deadlines, cancellation signals,
- * and other scoped values across API boundaries and between threads. Examples of functionality
- * propagated via context include:
+ * A context propagation mechanism which can carry scoped-values across API boundaries and between
+ * threads. Examples of state propagated via context include:
  * <ul>
- *   <li>Deadlines for a local operation or remote call.</li>
  *   <li>Security principals and credentials.</li>
- *   <li>Local and distributed tracing context.</li>
+ *   <li>Local and distributed tracing information.</li>
```

Comment:
Suggestions for how to document the intent are welcome. I used a hypothetical 'credential' in the example in the class heading as that's a very common case. I'm not sure that documenting the interceptor pattern here is the best idea.

Most of the code to do what you want is in the Contexts class. i.e. in your interceptor you would do....

class TracingInterceptor implements ServerInteceptor {
  <ReqT, RespT> ServerCall.Listener<ReqT> interceptCall(
      MethodDescriptor<ReqT, RespT> method,
      ServerCall<RespT> call,
      Metadata headers,
      ServerCallHandler<ReqT, RespT> next) {
      Trace trace = header.get(TRACE_KEY); // TRACE_KEY is a Metadata.Key
      return Contexts.interceptCall(
            Context.current().with(CONTEXT_TRACE_KEY, trace), // CONTEXT_TRACE_KEY is a Context.Key
            method, call, headers, next);
}